src/commands/build/watch/mod.rs

```diff
@@ -0,0 +1,82 @@
+mod watcher;
+pub use watcher::wait_for_changes;
+
+use crate::commands::build::{command, wranglerjs};
+use crate::commands::publish::Package;
+use crate::settings::project::{Project, ProjectType};
+use crate::terminal::message;
+use crate::{commands, install};
+
+use notify::{watcher, RecursiveMode, Watcher};
+use std::env;
+use std::sync::mpsc::{channel, Sender};
+use std::thread;
+use std::time::Duration;
+
+pub const COOLDOWN_PERIOD: Duration = Duration::from_millis(2000);
```

Comment:
Considering making this a config option, it refers to the amount of time wrangler will ignore new fs events when it detects the first one. Any new events during the cooldown period restart the timer.